### DISCUSSION TITLE
Offline support for getting document data

### DISCUSSION THREAD

**Discussion by morzel85** [capawesome-team/capacitor-firebase#713] [General]
I can get document data this way:

```
const { snapshot } = await FirebaseFirestore.getDocument({
  reference: 'someDocPathHere',
});
```

but when the app is offline it will give me such error: _FirebaseError: Failed to get document because the client is offline._

Is there a way to obtain most recently used document data (assume that I can preload the docs while being online)?
Can offline work cross-platform: on Web, Android and iOS?

Searching for "offline" on https://capawesome.io/plugins/firebase/cloud-firestore/ doesn't give me anything.
For "persistence" I only get "clearPersistence".

**Comment by robingenz** [comment]
Thank you for your question! Offline persistence (https://firebase.google.com/docs/firestore/manage-data/enable-offline) is not yet supported by the Capacitor Firebase Firestore plugin. Feel free to create a feature request. This should be easy to implement.

**Reply by morzel85** [reply]
Thanks for quick reply! Do you mean a GitHub Issue or an Idea in Discussions?

**Reply by robingenz** [reply]
GitHub issue

**Reply by rafal-castor** [reply]
hey, @robingenz do you mean that offline persistence is not enabled for web only, or for both web and native platforms?

**Reply by robingenz** [reply]
It's enabled by default on Android and iOS, see Firebase docs.

**Reply by morzel85** [reply]
So the gap here is just adding offline option switch for Web to wrapper configuration, and the `await FirebaseFirestore.getDocument` should just work for Android and iOS?

**Reply by robingenz** [reply]
Yes, but we should add this option to Android and iOS too in case someone wants to disable it.

**Reply by morzel85** [reply]
Issue added: https://github.com/capawesome-team/capacitor-firebase/issues/715
